salt/utils/xmlutil.py

```diff
@@ -165,6 +165,7 @@ def clean_node(parent_map, node, ignored=None):
         len(node.attrib.keys() - (ignored or [])) == 0
         and not list(node)
         and not has_text
+        and parent
```

Comment:
You should move that hunk into a separate commit. Adding a unit test in `test_xmlutil.py` for it would be nice too.